superset/utils/cache_manager.py

```diff
@@ -23,17 +23,17 @@ def __init__(self) -> None:
         super().__init__()
 
         self._cache = Cache()
-        self._tables_cache = Cache()
+        self._data_cache = Cache()
         self._thumbnail_cache = Cache()
 
     def init_app(self, app: Flask) -> None:
         self._cache.init_app(app, app.config["CACHE_CONFIG"])
-        self._tables_cache.init_app(app, app.config["TABLE_NAMES_CACHE_CONFIG"])
+        self._data_cache.init_app(app, app.config["DATA_CACHE_CONFIG"])
```

Comment:
Didn't add backward compatibility because the data objects put into this cache will become much much larger. If a Superset admin has configured in-memory cache for `TABLE_NAMES_CACHE_CONFIG`, they might run into problems. Therefore it's better to make things break to make them aware this change.

For most users, to migrate, it's as simple as renaming the config value.